cmd/kubeadm/app/util/staticpod/utils.go

```diff
@@ -165,6 +166,38 @@ func GetExtraParameters(overrides map[string]string, defaults map[string]string)
 	return command
 }
 
+// KustomizeStaticPod applies patches defined in kustomizeDir to a static Pod manifest
+func KustomizeStaticPod(pod v1.Pod, kustomizeDir string) (v1.Pod, error) {
```

Comment:
i think this function should accept a pointer to v1.Pod, so that we don't duplicate it on the stack on each function call.

it can also return a pointer to v1.Pod that is the new Pod after unmarshal.

Reply:
ok, I will do this in the implementation